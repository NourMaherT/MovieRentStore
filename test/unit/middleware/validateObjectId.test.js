const validateObjectId = require('../../../middleware/validateObjectId');
const mongoose = require('mongoose');

describe('ValidateObjectId middleware', () => {
    const scenarios = [
        {
          description: 'should return error 404 if object id is not vaild',
          error: 'Invalid ID',
          status: 404
        }
      ];
    scenarios.forEach(({ description, error, status }) => {
        it(description, async () => {
            const req = { 
                params: {
                    id: '1'
                }
            };
            const res = {
                send: jest.fn(),
                status: jest.fn(() => res)
              };
            const next = jest.fn();
            const validator = jest.fn().mockReturnValue(error);
    
            await validateObjectId(req, res, next);
            expect(res.status).toBeCalledWith(status);
            expect(res.send.mock.calls[0][0]).toMatchSnapshot();
        });
    });

});
