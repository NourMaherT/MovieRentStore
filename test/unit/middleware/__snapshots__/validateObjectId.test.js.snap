// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Validate middleware should return error 404 if object id is not vaild 1`] = `"Invalid ID"`;

exports[`ValidateObjectId middleware should return error 404 if object id is not vaild 1`] = `"Invalid ID"`;
